src/rust/zh.manhuaren/src/lib.rs

```diff
@@ -0,0 +1,512 @@
+#![no_std]
+extern crate alloc;
+use aidoku::{
+	error::Result,
+	prelude::*,
+	std::{format, print, ObjectRef, String, StringRef, Vec},
+	std::{
+		json,
+		net::{HttpMethod, Request},
+	},
+	Chapter, DeepLink, Filter, FilterType, Listing, Manga, MangaContentRating, MangaPageResult,
+	MangaStatus, MangaViewer, Page,
+};
+use alloc::vec;
+mod helper;
+
+const FILTER_GENRE: [i32; 30] = [
+	0, 31, 26, 1, 3, 27, 5, 2, 6, 8, 9, 25, 10, 11, 12, 17, 33, 37, 14, 15, 29, 20, 21, 4, 7, 30,
+	34, 36, 40, 61,
+];
+const FILTER_STATUS: [i32; 3] = [0, 1, 2];
+const FILTER_SORT: [i32; 3] = [0, 1, 2];
+
+const API_URL: &str = "http://mangaapi.manhuaren.com";
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut query = String::new();
+	let mut status: i32 = 0;
+	let mut genre: i32 = 0;
+	let mut sort: i32 = 10;
+
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				query = filter.value.as_string()?.read();
+			}
+			FilterType::Select => {
+				let index = filter.value.as_int()? as usize;
+				match filter.name.as_str() {
+					"连载状态" => {
+						status = FILTER_STATUS[index];
+					}
+					"分类" => {
+						genre = FILTER_GENRE[index];
+					}
+					_ => continue,
+				};
+			}
+			FilterType::Sort => {
+				let value = match filter.value.as_object() {
+					Ok(value) => value,
+					Err(_) => continue,
+				};
+				let index = value.get("index").as_int()? as usize;
+				sort = FILTER_SORT[index];
+			}
+			_ => continue,
+		}
+	}
+
+	print(format(format_args!("page: {}", page)));
+	print(format(format_args!("query: {}", query)));
+	print(format(format_args!("status: {}", status)));
+	print(format(format_args!("genre: {}", genre)));
+	print(format(format_args!("sort: {}", sort)));
+
+	if query.is_empty() {
+		get_manga_list_by_filter(
+			ListFilter {
+				status,
+				genre,
+				sort,
+			},
+			page,
+		)
+	} else {
+		get_manga_list_by_query(query, page)
+	}
+}
+
+#[get_manga_listing]
+fn get_manga_listing(_: Listing, _: i32) -> Result<MangaPageResult> {
+	print("get_manga_listing");
+
+	todo!()
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+	let mut args: Vec<(String, String)> = Vec::new();
+
+	args.push((String::from("mangaId"), id));
+
+	let qs = helper::generate_get_query(&mut args);
+
+	print("qs:");
+	print(&qs);
+
+	let url = String::from(API_URL) + "/v1/manga/getDetail?" + &qs;
+	print("url:");
+	print(&url);
+
+	let body = helper::request(url, HttpMethod::Get)?;
+	// print(&body);
+
+	let json = json::parse(body)?.as_object()?;
+	let manga = json.get("response").as_object()?;
+
+	let category_str = manga
+		.get("mangaTheme")
+		.as_string()
+		.unwrap_or_else(|_| StringRef::from(""))
```

Comment:
I think you can just do this?
```suggestion
		.unwrap_or(StringRef::from(""))
```